bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/obsrecord/TimeAccounting.scala

```diff
@@ -0,0 +1,50 @@
+package edu.gemini.spModel.obsrecord
+
+import edu.gemini.spModel.dataset.{ DatasetLabel, DatasetQaState }
+import edu.gemini.spModel.event.ObsExecEvent
+import edu.gemini.spModel.obsclass.ObsClass
+import edu.gemini.spModel.syntax.all._
+
+import scala.collection.JavaConverters._
+
+/**
+ * Time accounting calculation based on observation events.
+ */
+object TimeAccounting {
+
+  /**
+   * Calculate VisitTimes for the events in this visit.
+   */
+  def calc(
+    events: Vector[ObsExecEvent],
+    qa:     DatasetLabel => DatasetQaState,
+    oc:     DatasetLabel => ObsClass
+  ): VisitTimes = {
+
+    val ve = VisitEvents(events)
+
+    // Find the visit calculator that applies to the events.  We use the start
+    // time of the first event for the visit to select the corresponding
+    // calculator that is used for the rest of the sequence regardless of when
+    // it ends.
+    val calculator = for {
+      h <- ve.sorted.headOption
+      s <- h.site
+      c <- VisitCalculator.all.find(_.validAt(s).isBefore(h.instant))
+    } yield c
```

Comment:
This is where we keep old calculations using the old time accounting rules and allow new rules to be applied for new sequences.

Reply:
It seems like we might want to return something like `Either[Reason, VisitTimes]` so we can give a reason why we can't compute it (missing site, no applicable calculator), rather than returning an empty `VisitTimes` in that case. I kind of assume it's this way because that's how it has always worked, in which case ¯\\\_(ツ)\_/¯